data_structures/avltree.cpp

```diff
@@ -6,38 +6,48 @@
  * \warning This program is a poor implementation and does not utilize any of
  * the C++ STL features.
  */
-#include <algorithm>
-#include <iostream>
-#include <queue>
+#include <algorithm>  // for std::max
+#include <iostream>   // for std::cout
+#include <queue>      // for std::queue
 
-typedef struct node {
+using node = struct node {
     int data;
     int height;
     struct node *left;
     struct node *right;
-} node;
+};
 
-/** Create and return a new Node */
+/**
+ * @brief creates and returns a new node
+ * @return newly created node
```

Comment:
You might want to document the parameters as well.

Reply:
Please have a look at b535f5e4ad43ad6387c0959ade7c90969cea6dbb.